### ISSUE TITLE
Respect non-breaking spaces

### ISSUE THREAD

**Issue by chris5marsh** [beautify-web/js-beautify#869]
If there is a non-breaking space (`&nbsp;`) between two elements (and no whitespace), they should not be separated, i.e. whitespace shouldn't be added.

Source:

``` markup
<div><input type="checkbox" id="j" name="j" value="foo">&nbsp;<label for="j">Foo</label></div>
```

Expected result:

``` markup
<div>
    <input type="checkbox" id="j" name="j" value="foo">&nbsp;<label for="j">Foo</label>
</div>
```

Actual result:

``` markup
<div>
    <input type="checkbox" id="j" name="j" value="foo">&nbsp;
    <label for="j">Foo</label>
</div>
```

In this case the `&nbsp;` is there to keep the input and the label joined together so they will stay on the same line, but the same would be true of any situation where a non-breaking space is used without surrounding whitespace.

**Comment by chris5marsh** [comment]
I'll fork the repo and see if I can fix it...

**Comment by bitwiseman** [comment]
Hm... what happens if you put these tags inside a `<span>`?

**Comment by bitwiseman** [comment]
This is another issue related to #841.  Unless the beautifier understands inline elements it can't reason properly about when its allowed to insert newlines.  There might be a way to hack reasonable behavior in though.

**Comment by bitwiseman** [comment]
Should be fixed in 1.8.0-rc6.
